improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -752,8 +752,13 @@ def _location_and_scale_parameters_to_percentiles(
                 nans and these nans are not caused by a scale parameter of
                 zero.
         """
-        location_parameter_data = location_parameter.data.flatten()
-        scale_parameter_data = scale_parameter.data.flatten()
+        # Define a mask to be reapplied later
+        mask = capture_mask(location_parameter.data, scale_parameter.data)
```

Comment:
Yes that seems correct, and that code should work. I will change it. The reason I wrote a function called capture mask is because it is also used in probabilities and so I could apply to both (and elsewhere if we ever need to "capture" a mask?